src/Microsoft.SqlTools.ServiceLayer/Rename/RenameUtils.cs

```diff
@@ -0,0 +1,74 @@
+            }
+            if (String.IsNullOrEmpty(requestParams.TableInfo.TableName) || String.IsNullOrEmpty(requestParams.TableInfo.Database) || String.IsNullOrEmpty(requestParams.TableInfo.Schema) || String.IsNullOrEmpty(requestParams.TableInfo.OldName) || String.IsNullOrEmpty(requestParams.ChangeInfo.NewName))
+            {
+                throw new ArgumentException(SR.RenameRequestParametersNotNullOrEmpty);
+            }
+            if (!Regex.IsMatch(requestParams.TableInfo.Schema, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.Database, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.OldName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.TableInfo.TableName, RegexMatchPatternValidObjectName) || !Regex.IsMatch(requestParams.ChangeInfo.NewName, RegexMatchPatternValidObjectName))
+            {
+                throw new ArgumentOutOfRangeException(SR.NotAllowedCharacterForRenaming);
+            }
+        }
+        /// <summary>
+        /// Method to get the sqlobject, which should be renamed
+        /// </summary>
+        /// <param name="requestParams">parameters which are required for the rename operation</param>
+        /// <param name="connection">the sqlconnection on the server to search for the sqlobject</param>
+        /// <returns>the sqlobject if implements the interface IRenamable, so they can be renamed</returns>
+        public static IRenamable GetSQLRenameObject(ProcessRenameEditRequestParams requestParams, SqlConnection connection)
+        {
+            ServerConnection serverConnection = new ServerConnection(connection);
+            Server server = new Server(serverConnection);
+
+            Database database = new Database(server, requestParams.TableInfo.Database);
+            SqlSmoObject dbObject = database.Parent.GetSmoObject(new Urn(GetURNFromDatabaseSqlObjects(requestParams, serverConnection.TrueName)));
+
+            return (IRenamable)dbObject;
+        }
+        /// <summary>
+        /// Method to generate the Uniform Resource Name (URN) of a sqlobject 
+        /// </summary>
+        /// <param name="requestParams">parameters which are required for the rename operation</param>
+        /// <param name="trueNameOfServer">return the name of the server (@@Servername)</param>
+        /// <returns>URN string of the sqlobject</returns>
+        public static string GetURNFromDatabaseSqlObjects(ProcessRenameEditRequestParams requestParams, string trueNameOfServer)
+        {
+            String urnTableReplaced = String.Format(RenameUtils.UrnPrefix, trueNameOfServer, requestParams.TableInfo.Database, requestParams.TableInfo.TableName, requestParams.TableInfo.Schema, requestParams.TableInfo.OldName);
+
+            if (requestParams.ChangeInfo.Type == ChangeType.COLUMN)
+            {
+                return String.Format(urnTableReplaced + "/Column[@Name='{0}']", requestParams.TableInfo.OldName);
```

Comment:
>Column[@Name='{0}']", requestParams.TableI [](http://example.com/codeflow?start=58&length=42)

if you are building Urns manually you need to urn escape the the parameters.

Reply:
Ohh i did not think of that. Something like that?
```C# 
return String.Format(urnTableReplaced + @"/Column[@Name='{0}']", requestParams.TableInfo.OldName.ToString());
```
Do you think that is enough or should I add something specific?

Reply:
look at `Urn.EscapeString`

Reply:
Perfectly, thanks I will add it 👍